pootle/apps/pootle_app/management/commands/update_tmserver.py

```diff
@@ -73,7 +57,121 @@ def _parse_translations(self, **options):
                 'store__translation_project__language__code'
             ).order_by()
 
-        total = units_qs.count()
+        return units_qs.iterator(), units_qs.count()
+
+    def get_unit_data(self, unit):
+        """Return dict with data to import for a single unit."""
+        fullname = (unit['submitted_by__full_name'] or
+                    unit['submitted_by__username'])
+
+        email_md5 = None
+        if unit['submitted_by__email']:
+            email_md5 = md5(unit['submitted_by__email']).hexdigest()
+
+        return {
+            "_index": self.INDEX_NAME,
+            "_type": unit['store__translation_project__language__code'],
+            "_id": unit['id'],
+            'revision': int(unit['revision']),
+            'project': unit['store__translation_project__project__fullname'],
+            'path': unit['store__pootle_path'],
+            'username': unit['submitted_by__username'],
+            'fullname': fullname,
+            'email_md5': email_md5,
+            'source': unit['source_f'],
+            'target': unit['target_f'],
+        }
+
+
+class FileParser(object):
+
+    def __init__(self, options, *args, **kwargs):
+        super(FileParser, self).__init__(*args, **kwargs)
+        self.INDEX_NAME = kwargs.pop('index', None)
+        self.target_language = kwargs.pop('language', None)
+        self.project = kwargs.pop('project', None)
+
+    def get_units(self, filenames):
+        """Gets the units to import and its total count."""
+        units = []
+
+        for filename in filenames:
+            store = factory.getobject(filename)
+            if not store.gettargetlanguage() and not self.target_language:
+                raise CommandError("Unable to determine target language for "
+                                   "'%s'. Try again specifying a fallback "
+                                   "target language with --target-language" %
+                                   filename)
+
+            self.filename = filename
+            units.extend([unit for unit in store.units if unit.istranslated()])
+
+        return units, len(units)
+
+    def get_unit_data(self, unit):
+        """Return dict with data to import for a single unit."""
+        target_language = unit.gettargetlanguage()
+        if target_language is None:
+            target_language = self.target_language
+
+        return {
+            "_index": self.INDEX_NAME,
+            "_type": target_language,
+            "_id": unit.getid(),
+            'revision': 0,
+            'project': self.project,
+            'path': self.filename,
+            'username': 'anonymous',
```

Comment:
@julen I guess depends on what the UI shows.  But since this isn't Pootle's `nobody` I'm not sure it makes sense.  But I don't think ``anonymous`` is good either as I'm not sure what will display.  My guess is that it probably should be `None` since we don't know and the UI should do the right thing.